lfsapi/ntlm.go

```diff
@@ -9,10 +9,21 @@ import (
 	"net/url"
 	"strings"
 
-	"github.com/ThomsonReutersEikon/go-ntlm/ntlm"
 	"github.com/git-lfs/git-lfs/errors"
 )
 
+type ntmlCredentials struct {
+	domain   string
+	username string
+	password string
+}
+
+// Defines if the default credentials of the currently logged in user
+// should be used.
+func (c ntmlCredentials) IsDefaultCredentials() bool {
+	return c == ntmlCredentials{}
```

Comment:
Going off of the above, I think that it might be clearer to define the body as:

```go
return c.domain == "" && c.username == "" && c.password == ""
```